tests/test_Integration.py

```diff
@@ -4,123 +4,82 @@
 import requests
 from todo_app import app
 from dotenv import load_dotenv, find_dotenv
+import mongomock
+import pymongo
+
 
 @pytest.fixture
 def client():
     file_path = find_dotenv('.env.test')
     load_dotenv(file_path, override=True)
 
-    test_app = app.create_app()
-
-    with test_app.test_client() as client:
-        yield client
+    with mongomock.patch(servers=(('fakemongo.com', 27017),)):
+        test_app = app.create_app()
+        with test_app.test_client() as client:
+            yield client
 
-class StubResponse():
-    def __init__(self, fake_response_data):
-        self.fake_response_data = fake_response_data
+def test_index_page(client):
+    collection = pymongo.MongoClient(os.getenv("COSMOS_CONNECTION_STRING"))
+    database = os.getenv("DATABASE")
     
-    def json(self): 
-        return self.fake_response_data
+    collection[database].cards.insert_many(sample_card_lists_response)
     
-def stub(url, params):
-    test_board_id = os.environ.get('TRELLO_BOARD_ID')
-    fake_response_data = None
-
-    if url == f'https://api.trello.com/1/boards/{test_board_id}/lists':
-        fake_response_data = sample_trello_lists_response
-        return StubResponse(fake_response_data)
-    
-    if url == f'https://api.trello.com/1/cards':
-        fake_response_data = create_trello_card
-        return StubResponse(fake_response_data)
+    response = client.get('/')
     
-    raise Exception(f'Integration test stub no mock for url "{url}"')
+    assert response.status_code == 200
+    assert 'Task 2' in response.data.decode()
 
 
-def test_index_page(monkeypatch, client):
-    monkeypatch.setattr(requests, 'get', stub)
-    response = client.get('/')
+def test_add_item(client):
+    collection = pymongo.MongoClient(os.getenv("COSMOS_CONNECTION_STRING"))
+    database = os.getenv("DATABASE")
 
-    assert response.status_code == 200
-    assert 'Task 2' in response.data.decode()
+    collection[database].cards.insert_many(sample_card_lists_response)
+
+    response = client.get('/')
 
-def test_add_item(monkeypatch, client):
-    monkeypatch.setattr(requests, 'get', stub)
-    monkeypatch.setattr(requests, 'post', stub)
+    collection[database].cards.insert_one(create_card)
```

Comment:
I don't quite see what the point of this line is - it adds a new thing to the DB that we promptly ignore?